examples/client.rs

```diff
@@ -49,14 +48,14 @@ fn main() {
     let tcp_handle = sockets.add(tcp_socket);
 
     {
-        let socket: &mut TcpSocket = sockets.get_mut(tcp_handle).as_socket();
+        let mut socket = sockets.get::<TcpSocket>(tcp_handle).unwrap();
```

Comment:
I believe dispatching over the socket type is the rare case (it only happens inside smoltcp and in OS/firmware code that's generic over the socket, e.g. fd handling), so let's have a panicking .get() and a non-panicking get_opt(), seems cleaner than unwrapping everywhere.